### PR TITLE
Add <Radio /> Component

### PR DESCRIPTION
**Pull request by jackieo5023**
* radio group props always override child if there's a conflict
* basically the user should always use `Radio.Group` since it does a lot of the weird HTML stuff for them - in a normal use case, they also would only need to set the `id` and `value` props in `Radio` and `name` in `Radio.Group`
   * This would change if they decide to use `onChange` and `value`, but again, this should all be handled with `Radio.Group`

I can add more documentation to this later, but for now I just pasted my examples as jsx into the page.

### CODE REVIEW

**Comment by tko22** [comment]
hm yeah maybe we can make it pull from colors in a radio theme. This way you can customize it easily. 

do you what happens when you pass a `color` prop into Radio? since that prop is in COMMON but it’s only active in label base?

**Comment by jackieo5023** [comment]
> hm yeah maybe we can make it pull from colors in a radio theme. This way you can customize it easily.
> 
> do you what happens when you pass a `color` prop into Radio? since that prop is in COMMON but it’s only active in label base?

That will change the label color - however, I could probably do something to intercept that and make it change the button color instead.  The other option I guess would be what you said to do and put it all into the radio theme.

**Comment by tko22** [comment]
yeah i think radio theme would work best in that case

File: src/components/TextField/TextField.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ const TextFieldSuffixIcon = styled(Icon)`
 
 const TextFieldBase = styled.input`
   ${get('typography.forms')}
-  color: ${get('colors.text.coolTone.main')};
+  color: ${get('colors.text.grayScale.main')};
```
Comment (tko22) [inline src/components/TextField/TextField.js:57]: tyty

**Comment by jackieo5023** [comment]
wait tim i still haven't changed anything and this is still a draft HAHA

I should have time tomorrow to implement changes and write some docs

**Comment by tko22** [comment]
woops haha no worries

File: docs/pages/components/radio.mdx [hunk 2]
```diff
@@ -0,0 +1,119 @@
+export const meta = { type: "components", title: "Radio" };
+
+import Layout from "../../components/layout.js";
+export default Layout({ ...meta });
+
+Props from Radio.Group that also appear in Radio will be overridden by Radio.Group's. In general, you should never use `Radio` without a `Radio.Group`.
+
+Jackie's code dump of examples that can be cleaned up later:
```
Comment (tko22) [inline docs/pages/components/radio.mdx:8]: dope

File: src/__tests__/Radio.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+  it('passes down "disabled" prop to children', () => {
+    const group = renderJSON(
+      <Radio.Group disabled>
+        <Radio id="1">option 1</Radio>
+        <Radio id="2">option 2</Radio>
+        <Radio id="3">option 3</Radio>
+      </Radio.Group>,
+    );
+    group.children.map((child) => expect(child.props.disabled).toBeTruthy());
```
Comment (tko22) [inline src/__tests__/Radio.js:85]: ohh shit

File: src/components/Radio/Radio.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+);
+
+Radio.defaultProps = { theme, error: 'default' };
+
+Radio.propTypes = {
+  checked: PropTypes.bool,
+  children: PropTypes.node,
+  defaultChecked: PropTypes.bool,
+  disabled: PropTypes.bool,
+  error: PropTypes.oneOf(['default', 'error', 'warning', 'success']),
```
Comment (tko22) [inline src/components/Radio/Radio.js:124]: do you think this should be a different name? i was thinking `variant` or `state`. I was just a little confused on that but if you think it's fine, it's ok with me too cuz it'll be documented

Comment (tko22) [reply src/components/Radio/Radio.js:124]: oh wait `TextField` uses `error` prop, we should keep it this way then.

File: docs/pages/components/radio.mdx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+| :----------- | :--------------------------------------------------------- | :----------: | :---------------------------------------------------------------------------------------------------------------------------- |
+| children     | Node                                                       |              | These should be `Radio` components to be grouped together                                                                     |
+| defaultValue | Any                                                        |              | The default value (out of the children `Radio`s) to be selected. This should not be used when `value` and `onChange` are used |
+| disabled     | Boolean                                                    |    false     | Disable input for all children                                                                                                |
+| error        | `default` &#124; `error` &#124; `warning` &#124; `success` |  'default'   | Set the error state for all children                                                                                          |
```
Comment (tko22) [inline docs/pages/components/radio.mdx:114]: `&#124;` is genius imma change all the docs to that

File: src/components/Radio/Radio.js (hunk 4 above)
Comment (jackieo5023) [reply src/components/Radio/Radio.js:124]: sure, we can always change it later if that would make both more clear

**Pull request merged**